crates/ty_python_semantic/src/types.rs

```diff
@@ -6895,6 +6938,46 @@ impl<'db> FunctionType<'db> {
         }
     }
 
+    fn is_subtype_of(self, db: &'db dyn Db, other: Self) -> bool {
+        // A function literal is the subtype of itself, and not of any other function literal.
+        // However, our representation of a function literal includes any specialization that
+        // should be applied to the signature. Different specializations of the same function
+        // literal are only subtypes of each other if they result in the same signature. (Note that
+        // the equality check above will have already handled the case where `self` and `target`
+        // are the same function literal with the same specialization.)
+        self.body_scope(db) == other.body_scope(db)
+            && self
+                .into_callable_type(db)
+                .is_subtype_of(db, other.into_callable_type(db))
+    }
+
+    fn is_assignable_to(self, db: &'db dyn Db, other: Self) -> bool {
+        // A function literal is assignable to itself, and not to any other function literal.
+        // However, our representation of a function literal includes any specialization that
+        // should be applied to the signature. Different specializations of the same function
+        // literal are only assignable to each other if they result in the same signature. (Note
+        // that the equality check above will have already handled the case where `self` and
+        // `target` are the same function literal with the same specialization.)
```

Comment:
Same here, it would be useful to specify which equality check this is referring to.

Reply:
Ditto above